core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -703,37 +704,52 @@ object DeltaTable {
   }
 
   /**
-   * Check if the provided `identifier` string, in this case a file path,
-   * is the root of a Delta table using the given SparkSession.
+   * Check if the provided `identifier` string represents a Delta table.
+   * Identifier strings can be either a path, table or view name.
+   * In the case of a file path, it checks if it is the root of a Delta
+   * table using the given SparkSession.
    *
-   * An example would be
+   * Examples are as follows:
    * {{{
    *   DeltaTable.isDeltaTable(spark, "path/to/table")
+   *   DeltaTable.isDeltaTable(spark, "MyDeltaTable")
    * }}}
    *
    * @since 0.4.0
    */
   def isDeltaTable(sparkSession: SparkSession, identifier: String): Boolean = {
-    val identifierPath = new Path(identifier)
-    if (sparkSession.sessionState.conf.getConf(DeltaSQLConf.DELTA_STRICT_CHECK_DELTA_TABLE)) {
-      val rootOption = DeltaTableUtils.findDeltaTableRoot(sparkSession, identifierPath)
-      rootOption.isDefined && DeltaLog.forTable(sparkSession, rootOption.get).tableExists
-    } else {
-      DeltaTableUtils.isDeltaTable(sparkSession, identifierPath)
+    val tableId: Try[TableIdentifier] =
+      Try(sparkSession.sessionState.sqlParser.parseTableIdentifier(identifier))
+    val isDeltaTableByName =
+      tableId.isSuccess && DeltaTableUtils.isDeltaTable(sparkSession, tableId.get)
+
+    val isDeltaTableByPath: Boolean = {
```

Comment:
you dont need to compute this if `isDeltaTableByName` is true already .. right? 
so it might better to make this lazy .. either a lazy val or a def.